pkg/kubelet/cm/types.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+// ResourceConfig holds information about all the supported cgroup resource parameters.
+type ResourceConfig struct {
+	// Memory limit (in bytes).
+	Memory *int64
+	// CPU shares (relative weight vs. other containers).
+	CpuShares *int64
+	// CPU hardcap limit (in usecs). Allowed cpu time in a given period.
+	CpuQuota *int64
+}
+
+// CgroupConfig holds the cgroup configuration information.
+// This is common object which is used to specify
+// cgroup information to both systemd and raw cgroup fs
+// implementation of the Cgroup Manager interface.
+type CgroupConfig struct {
+	// We would expect systemd implementation to make appropriate
+	// name conversion. For example, if we pass /foo/bar
+	// then systemd should convert the name to something like
+	// foo.slice/foo-bar.slice
+
+	// Fully qualified name
+	Name string
```

Comment:
Yeah. That's fine. But mention that behavior explicitly.

On Thu, Jun 23, 2016 at 4:16 PM, Buddha Prakash <notifications@github.com>
wrote:

> In pkg/kubelet/cm/types.go
> <https://github.com/kubernetes/kubernetes/pull/27755#discussion_r68331833>
> :
>
> > +	// CPU hardcap limit (in usecs). Allowed cpu time in a given period.
> > +	CpuQuota *int64
> > +}
> > +
> > +// CgroupConfig holds the cgroup configuration information.
> > +// This is common object which is used to specify
> > +// cgroup information to both systemd and raw cgroup fs
> > +// implementation of the Cgroup Manager interface.
> > +type CgroupConfig struct {
> > +	// We would expect systemd implementation to make appropriate
> > +	// name conversion. For example, if we pass /foo/bar
> > +	// then systemd should convert the name to something like
> > +	// foo.slice/foo-bar.slice
> > +
> > +	// Fully qualified name
> > +	Name string
>
> I was thinking of just creating the leaf cgroups. I assume that we wont be
> passing paths where the parent directory doesn't already exist. Does that
> sound good?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27755/files/9b264a0fe89b04d775a58996df07e135b030cec6..773b7e7144cb41cbb65d45b4e902e3888636af17#r68331833>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvIKBRx1w6L7n3xpS-YdLs-QlqCMBqRks5qOxPTgaJpZM4I6XXz>
> .
>